go.mod

```diff
@@ -5,27 +5,27 @@ go 1.18
 require (
 	github.com/google/go-cmp v0.5.9
 	github.com/grpc-ecosystem/grpc-gateway/v2 v2.15.0
-	google.golang.org/grpc v1.51.0
-	google.golang.org/protobuf v1.28.1
+	google.golang.org/grpc v1.56.3
+	google.golang.org/protobuf v1.30.0
```

Comment:
do we need to change the protobuf version to address the CVE? If not, I would prefer doing that in a separate PR since there are more changes in tail, like updating the Dockerfile and regenerating the proto stubs, e.g. https://github.com/kserve/modelmesh-runtime-adapter/pull/58

Reply:
yes, protobuf is affected as well

Reply:
oh, sry, protobuf isn't affected, but the grpc version depends on that protobuf version, that's why it was updated (go get).

Reply:
Right `go get ...` versus `replace ... ` and `go mod tidy -compat=1.18`

I would start by removing the indirect dependencies from `go.mod`, update the direct dependencies `google.golang.org/grpc v1.56.3` and add a replace section for `golang.org/x/net v0.17.0` and `k8s.io/apimachinery v0.27.0` to tackle #33 

Initial contents of `go.mod`:
```go.mod
module github.com/kserve/rest-proxy

go 1.18

require (
	github.com/google/go-cmp v0.5.9
	github.com/grpc-ecosystem/grpc-gateway/v2 v2.15.0
	google.golang.org/grpc v1.56.3
	google.golang.org/protobuf v1.28.1
	sigs.k8s.io/controller-runtime v0.14.1
)

replace (
	golang.org/x/net => golang.org/x/net v0.17.0
	k8s.io/apimachinery => k8s.io/apimachinery v0.27.0
)
```

Now run `go mod tidy -compat=1.18` ...

Resulting `go.mod`:

```go.mod
module github.com/kserve/rest-proxy

go 1.18

require (
	github.com/google/go-cmp v0.5.9
	github.com/grpc-ecosystem/grpc-gateway/v2 v2.15.0
	google.golang.org/grpc v1.56.3
	google.golang.org/protobuf v1.30.0
	sigs.k8s.io/controller-runtime v0.14.1
)

require (
	github.com/go-logr/logr v1.2.3 // indirect
	github.com/go-logr/zapr v1.2.3 // indirect
	github.com/gogo/protobuf v1.3.2 // indirect
	github.com/golang/protobuf v1.5.3 // indirect
	github.com/google/gofuzz v1.1.0 // indirect
	github.com/json-iterator/go v1.1.12 // indirect
	github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd // indirect
	github.com/modern-go/reflect2 v1.0.2 // indirect
	github.com/rogpeppe/go-internal v1.11.0 // indirect
	go.uber.org/atomic v1.7.0 // indirect
	go.uber.org/multierr v1.6.0 // indirect
	go.uber.org/zap v1.24.0 // indirect
	golang.org/x/net v0.10.0 // indirect
	golang.org/x/sys v0.13.0 // indirect
	golang.org/x/text v0.13.0 // indirect
	google.golang.org/genproto v0.0.0-20230410155749-daa745c078e1 // indirect
	gopkg.in/inf.v0 v0.9.1 // indirect
	gopkg.in/yaml.v2 v2.4.0 // indirect
	k8s.io/apimachinery v0.26.0 // indirect
	k8s.io/klog/v2 v2.90.1 // indirect
	k8s.io/utils v0.0.0-20230209194617-a36077c30491 // indirect
	sigs.k8s.io/json v0.0.0-20221116044647-bc3834ca7abd // indirect
	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
)

replace (
	golang.org/x/net => golang.org/x/net v0.17.0
	k8s.io/apimachinery => k8s.io/apimachinery v0.27.0
)
```

Reply:
I see, thanks for the explanation.
It seems this approach is better :)